ssl/record/ssl3_record.c

```diff
@@ -1335,6 +1354,9 @@ int n_ssl3_mac(SSL *ssl, SSL3_RECORD *rec, unsigned char *md, int sending)
     if (!sending &&
         EVP_CIPHER_CTX_mode(ssl->enc_read_ctx) == EVP_CIPH_CBC_MODE &&
         ssl3_cbc_record_digest_supported(hash)) {
+#ifdef OPENSSL_NO_DEPRECATED_3_0
+        return 0;
```

Comment:
It's a little weird that the commit message talks about MD5/SHA1 but the helper claims that SHA-2 is supported, but IIUC this is just used for sslv3 and thus the SHA-2 support is vestigal